app/src/main/java/org/schabi/newpipe/player/helper/PlayerDataSource.java

```diff
@@ -17,9 +18,12 @@
 import com.google.android.exoplayer2.upstream.TransferListener;
 
 public class PlayerDataSource {
+
+    public static final int LIVE_STREAM_EDGE_GAP_MILLIS = 10000;
+
+    private static final double PLAYLIST_STUCK_TARGET_DURATION_COEFFICIENT = 15;
```

Comment:
Ah short documentation what this values does and why it was chosen would be great 😄

Reply:
> why it was chosen

It's more like by testing on (ultra)low latency livestreams but on livestreams with an higher segment target duration, the coefficient maybe too big.

What's your advice (and @Redirion's one)?

Reply:
> What's your advice

Good question. I'm not really technically into this.

Note: The formula where this is used is:
```
if (currentTimeMs - lastSnapshotChangeMs > C.usToMs(playlistSnapshot.targetDurationUs) * playlistStuckTargetDurationCoefficient) {
   // do throw error
}
```
https://github.com/google/ExoPlayer/blob/029a2b27cbdc27cf9d51d4a73ebeb503968849f6/library/hls/src/main/java/com/google/android/exoplayer2/source/hls/playlist/DefaultHlsPlaylistTracker.java#L728-L733

Do you have any data regarding the values above that might helps choosing a value?

Anyway the current 15 would be ~4,3x greater then the default (3.5) so I think we could say/document "Choose a 4-times greater value than the default to ensure that livestreams which got stuck for a moment don't crash/fail"

Reply:
I will use the following JavaDoc:

```java
    /**
     * An approximately 4.3 times greater value than the
     * {@link DefaultHlsPlaylistTracker#DEFAULT_PLAYLIST_STUCK_TARGET_DURATION_COEFFICIENT default}
     * to ensure that (very) low latency livestreams which got stuck for a moment don't crash too
     * early.
     */
```